content/entry-invalidation/index.md

```diff
@@ -0,0 +1,139 @@
+---
+rfc:
+start_date: 2018-02-12
+pr:
+status: draft
+---
+
+# Entry Invalidation
+
+## Summary
+
+This RFC proposes a way to invalidate an entry so previous changes considered
+mistakes can be removed from the computed list of records.
+
+
+## Motivation
+
+Currently, the append-only nature of Registers doesn't cope well with
+mistakes. A situation that has arisen more than once is getting two records
+about the same thing with different identifiers because a human mistake.
+Another situation is having a change in the history of a record that is wrong.
+
+We need a way to mark entries as invalid so we can keep compatibility with
+other copies of the Register (consumers, replicas, etc) and allow computing
+the correct(ed) list of records.
+
+
+## Explanation
+
+[TODO: Currently two alternatives are explored]
+
+
+### Alternative A: Special entry
+
+The proposal introduces a new type of entry that allows listing a set of entry
+identifiers. This mechanism would allow invalidating discrete entries and by
+extension invalidating the full history for a key when all entries for that
+key are flagged as invalid. It also introduces a new RSF command to avoid
+overloading the `append-entry` command.
+
+[TODO: entry number or entry hash? Entry number is aligned with existing ways
+to refer to entries. Entry hash invalidates the entry by its content]
+
+#### RSF
+
+```
+invalidate-entry	user	2018-02-12T10:11:12Z	3;234;355
+```
+
+#### JSON
+
+```json
+{
+  "entry-number": "356",
+  "revoked-entries": ["3", "234", "355"],
+  "entry-timestamp": "2018-02-12T10:11:12Z"
+}
+```
+
+#### Properties
+
+* New entry type without key. Breaking change diverging from previous designs.
+* Records are unaffected.
+* Revoked entries require a mechanism/documentation to explain how to apply
+  them when computing the list of records.
+* Entry proofs are kept intact and invalidations are part of the tree.
+
```

Comment:
Thanks for the thorough review, I have to re-read what I wrote. It's been a while. In any case there is another case (which is why this is important sooner rather than latter):

An entry introduces a record under a new key but it should've been an update to an existing key.

That means the entry needs to be marked as "ignore this one, mistake", a start/end date is the wrong level of abstraction to flag this. It's not about the data, it's about the metadata.

In any case, I'll revisit this soon and reply to your comments.